src/main/frontend/src/components/AddReleaseYearForm.js

```diff
@@ -4,10 +4,138 @@
 //
 
 class AddReleaseYearForm extends React.Component {
-	
+	constructor(props) {
+		super(props);
+		this.state = {
+			year: null,
+			hasServerError: false,
+			validationErrors: [],
+			isDisabled: true,
+		};
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleChange = this.handleChange.bind(this);
+		this.generateRange = this.generateRange.bind(this);
+	}
+
+	generateRange(start, end) {
+		return new Array(end - start + 1)
+			.fill()
+			.map((_, idx) => start + idx);
+	}
+
+	handleChange(event) {
+		event.preventDefault();
+		this.setState({
+			year: event.target.value,
+			isDisabled: false
+		});
+	}
+
+	handleSubmit(event) {
+		event.preventDefault();
+
+		this.setState({
+			isDisabled: true,
+			hasServerError: false,
+			validationErrors: []
+		});
+
+		axios
+			.patch(
+				this.props.url,
+				[
+					{
+						op: 'add',
+						path: '/release_year',
+						value: this.state.year
+					}
+				],
+				{
+					headers: {
+						[this.props.csrfHeaderName]: this.props.csrfTokenValue,
+						'Cache-Control': 'no-store'
+					},
+					validateStatus: status => {
+						return status == 204 || status == 400;
+					}
+				}
+			)
+			.then(response => {
+				const data = response.data;
+				if (data.hasOwnProperty('fieldErrors')) {
+					const fieldErrors = [];
+					if (data.fieldErrors.year) {
+						fieldErrors.push(...data.fieldErrors.year);
+					}
+					this.setState({
+						isDisabled: false,
+						validationErrors: fieldErrors
+					});
+					return;
+				}
+
+				// no need to reset the state as page will be reloaded
+				window.location.reload();
+			})
+			.catch(error => {
+				console.error(error);
+				this.setState({ isDisabled: false, hasServerError: true });
+			});
+	}
+
 	render() {
+		const rangeOfYears = this.generateRange(
+			this.props.sinceYear,
+			this.props.tillYear
+		);
+		const hasValidationErrors = this.state.validationErrors.length > 0;
 		return (
-			<div></div>
-		)
+			<div className="col-sm-12 form-group">
+				<form className="form-horizontal" onSubmit={this.handleSubmit}>
```

Comment:
In order to make an errored field to be highlighted properly we also should have `has-error` class on a form:
```suggestion
				<form className={`form-horizontal ${hasValidationErrors ? 'has-error' : ''}`} onSubmit={this.handleSubmit}>
```